packages/styled-components/src/sheet/test/HoistedTag.test.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+
+import { VirtualTag } from '../Tag';
+import { DefaultHoistedTag } from '../HoistedTag';
+
+describe('Works like a groupedTag if there are no hoisted rules', () => {
+  /* 
+  This set of tests is mostly copy/pasted from GroupedTag.test.js,
+  but since we need to dig into the normal/hoisted tags
+  it didn't make sense to try and reduce the duplication.
+   */
```

Comment:
Not sure if this comment is super needed, but I wanted to make sure that I called this copy/paste and adjust situation out in the PR. Do any of y'all know a better way to accomplish that? It's only 2 tests and ~60 lines, so I went with the most straightforward way to get it tested.

Reply:
I think that’s fine. We may want to just change the structure to be more functional / grouped a little nicer so this can be resolved nicely.
It’s one of the many signs that after the feature is done, the code will need some restructuring